hera_librarian/async_transfers/globus.py

```diff
@@ -67,34 +54,47 @@ def authorize(self, settings: "ServerSettings"):
         to specific endpoints. We will do our best to handle this as it comes up
         to provide the user with nicer error messages, though we may not have
         caught all possible failure modes.
+
+        Parameters
+        ----------
+        settings : ServerSettings object
+            The settings for the Librarian server. These settings should include
+            the Globus login information.
+
+        Returns
+        -------
+        Globus authorizer or None
+            The object returned will be an instance of
+            globus_sdk.RefreshTokenAuthorizer (if using the Native App),
+            globus_sdk.AccessTokenAuthorizer (if using the Confidential App),
+            or None (if we could not successfully authenticate).
         """
         if settings.globus_enable is False:
-            return False
-
-        if self.authorizer is None:
-            if settings.globus_client_native_app:
-                try:
-                    client = globus_sdk.NativeAppAuthClient(settings.globus_client_id)
-                    self.authorizer = globus_sdk.RefreshTokenAuthorizer(
-                        settings.globus_client_secret, client
-                    )
-                except globus_sdk.AuthAPIError as e:
-                    return False
-            else:
-                try:
-                    client = globus_sdk.ConfidentialAppAuthClient(
-                        settings.globus_client_id, settings.globus_client_secret
-                    )
-                    tokens = client.oauth2_client_credentials_tokens()
-                    transfer_tokens_info = tokens.by_resource_server[
-                        "transfer.api.globus.org"
-                    ]
-                    transfer_token = transfer_tokens_info["access_token"]
-                    self.authorizer = globus_sdk.AccessTokenAuthorizer(transfer_token)
-                except globus_sdk.AuthAPIError:
-                    return False
-
-        return True
+            return None
+
+        if settings.globus_client_native_app:
+            try:
+                client = globus_sdk.NativeAppAuthClient(settings.globus_client_id)
+                authorizer = globus_sdk.RefreshTokenAuthorizer(
+                    settings.globus_client_secret, client
```

Comment:
@plaplant is this the refresh token provided by Globus for the native app?

If yes, would you mind changing the call to have name-based assignment instead of positional? It will increase the readability significantly.

Something like:
```
 authorizer = globus_sdk.RefreshTokenAuthorizer(
                    refresh_token=settings.globus_client_secret,
                    client=client
)
```

Reply:
Thanks for the suggestion, I've moved all of the `globus_sdk` calls to use keyword arguments.